src/lib/shopping-fpti-event-publisher.js

```diff
@@ -0,0 +1,55 @@
+/* @flow */
+import type { FptiInput, Config, ContainerSummary } from '../types';
+
+import { trackFptiV2 } from './fpti';
+import { fetchContainerSettings } from './get-property-id';
+import { logger } from './logger';
+
+export const ShoppingEventPublisher = (config : Config) => {
+
+  const trackEventQueue = [];
+
+  function isAllowedToPublishEvent() : boolean {
+    const isContainerIdSet  = (config.propertyId !== undefined && config.propertyId !== null && config.propertyId.length > 0);
+    return isContainerIdSet;
+  }
+
+  function publishFptiEvent(fptiEvent : FptiInput) {
```

Comment:
minor : can you add a short comment on why some events are okay to be published immediately and some are queued ? - that we need to wait until the containerSummary is returned and that this queuing is done only at the beginning.